webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criterion-preview.js

```diff
@@ -16,52 +16,65 @@ export class SakaiRubricCriterionPreview extends RubricsElement {
             </div>
-            <div class="criterion-ratings">
-              <div class="cr-table">
-                <div class="cr-table-row">
-                ${c.ratings.map(r => html`
-                  <div class="rating-item" id="rating_item_${r.id}" >
-                    <h5 class="criterion-item-title">${r.title}</h5>
-                    <div class="div-description">
-                      <p>${r.description}</p>
+          ` : html`
+            <div id="criterion_row_${c.id}" class="criterion-row">
+              <div class="criterion-detail">
+                <h4 class="criterion-title">${c.title}</h4>
+                <p>${unsafeHTML(c.description)}</p>
+                ${this.weighted ? html`
+                    <div class="criterion-weight">
+                        <span>
+                          <sr-lang key="weight">Weight</sr-lang>
+                        </span>
+                        <span>${c.weight.toLocaleString(this.locale)}</span>
+                        <span>
+                          <sr-lang key="percent_sign">%</sr-lang>
+                        </span>
+                    </div>`
+                  : ""
+                }
+              </div>
+              <div class="criterion-ratings">
+                <div class="cr-table">
+                  <div class="cr-table-row">
+                  ${c.ratings.map(r => html`
+                    <div class="rating-item" id="rating_item_${r.id}" >
+                      <h5 class="criterion-item-title">${r.title}</h5>
+                      <div class="div-description">
+                        <p>${r.description}</p>
+                      </div>
+                      <span class="points">
+                        ${this.weighted && r.points > 0 ? html`
+                            <b>
+                              (${parseFloat((r.points * (c.weight / 100)).toFixed(2)).toLocaleString(this.locale)})
+                            </b>`
+                          : ""
+                        }
+                        ${r.points.toLocaleString(this.locale)}
+                        <sr-lang key="points">Points</sr-lang>
+                      </span>
                     </div>
-                    <span class="points">
-                      ${this.weighted && r.points > 0 ? html`
-                          <b>
-                            (${parseFloat((r.points * (c.weight / 100)).toFixed(2)).toLocaleString(this.locale)})
-                          </b>`
-                        : ""
-                      }
-                      ${r.points.toLocaleString(this.locale)}
-                      <sr-lang key="points">Points</sr-lang>
-                    </span>
+                  `)}
                   </div>
-                `)}
                 </div>
               </div>
             </div>
-          </div>
+          `}
         `)}
       </div>
     `;
   }
+
+  isCriterionGroup(criterion) {
+    return criterion.ratings.length == 0;
```

Comment:
===

Reply:
> Mainly small stuff. Generally, this is a nicely done change. Thanks.

Generally, when I am working on a feature like this I am trying to stick to present patterns in the code. Most of what you commented on is STRONGLY inspired (pretty much a copy), by the existing code, which seems to be authored by you. I think it is fine if you change your mind about how that should be written. But let's please not rethink everything once we want to add a feature. I think this changes should be seperate, and are not related to the feature.

TLDR: Let's focus on what functionality is added, not on which patterns are inherited from the existing codebase.